### PR TITLE
Add the accumulate problem

### PR DESCRIPTION
**Pull request by sturmer**
Getting my feet wet. Any feedback would be appreciated :)

### CODE REVIEW

**Comment by yurrriq** [comment]
I think the general idea is to avoid built-in functions like `map`.

You could implement `accumulate` using pattern-matching like this:
```racket
(define/match (accumulate proc lst)
  [(proc '())          '()]
  [(proc `(,x . ,rst)) `(,(proc x) . ,(accumulate proc rst))])
```

or more traditionally like this:
```racket
(define (accumulate proc lst)
  (if (null? lst)
      '()
      (cons (proc (car lst))
            (accumulate proc (cdr lst)))))
```

File: accumulate/accumulate-test.rkt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  (define suite
+    (test-suite
+     "accumulate tests"
+     
+     (test-equal? "empty list"
+                  (accumulate empty (lambda (arg) (arg)))
+                  empty)     
+     
+     (test-equal? "doubling"
+                  (accumulate '(1 3 4) (lambda (arg) (arg * arg)))
```
Comment (yurrriq) [inline accumulate/accumulate-test.rkt]: I'm assuming you meant `(* arg arg)` here.

File: accumulate/accumulate-test.rkt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+(module+ test
+  (require rackunit rackunit/text-ui)
+  
+  (define suite
+    (test-suite
+     "accumulate tests"
+     
+     (test-equal? "empty list"
+                  (accumulate empty (lambda (arg) (arg)))
```
Comment (yurrriq) [inline accumulate/accumulate-test.rkt]: The `(arg)` in the body of the `lambda` here, attempts to call it as a function with no arguments, which surely isn't your intention.

**Comment by yurrriq** [comment]
It would probably be helpful to have a couple more tests. You might find some inspiration in [the LFE version][1].

[1]: https://github.com/exercism/xlfe/blob/master/accumulate/test/accumulate-tests.lfe

**Comment by yurrriq** [comment]
Sorry if my comments come off as terse or overly critical. That's not my intention. It's late here an I shouldn't be coding any more tonight...

Thanks very much for getting involved! I hope my feedback is constructive rather than discouraging.

**Comment by sturmer** [comment]
Hi @yurrriq no worries, your comments are very constructive and I'll address them carefully later today (it was very early here and I missed silly stuff like ``(* arg arg)`` instead of ``(arg * arg)`` :-) )

File: accumulate/accumulate-test.rkt (hunk 1 above)
Comment (bennn) [inline accumulate/accumulate-test.rkt]: or `(arg . * . arg)`

**Comment by sturmer** [comment]
I have addressed the immediate concerns. I am trying to reproduce also the last test from LFE but I am still a 3-year-old when it comes to list comprehensions (I am using this chance to learn that too). Thanks again for your comments @yurrriq and @bennn, please go ahead if you have other comments.

**Comment by sturmer** [comment]
Hi, anyone has any other comments? If the initial comments are addressed, I'd be happy to know how I can help to have it merged.

**Comment by yurrriq** [comment]
Without running the code it looks good to me. @arguello, will you check it out and merge?

**Comment by arguello** [comment]
looks good, thank you.

**Comment by yurrriq** [comment]
:tada:

**Comment by sturmer** [comment]
Whooo thanks @arguello and @yurrriq!

**Pull request merged**